src/main/java/org/eclipse/opensmartclide/dbapi/model/Deployment.java

```diff
@@ -23,12 +23,13 @@ public class Deployment {
     @NotNull
     private String user_id;
 
+    @NotNull
     private String user;
 
     //@DocumentReference
     //private User user;
 
-    //@NotNull
+    @NotNull
```

Comment:
I think that the NotNull annotation is not enough here.

Actually, NotNull means that a field cannot be null, but it can be empty. We should ensure that the field should not be empty as well.

So, I think it is better to use the NotEmpty annotation.

From [here](https://docs.jboss.org/hibernate/beanvalidation/spec/2.0/api/):

Annotation Type NotEmpty: The annotated element must not be null nor empty.